packages/core/docs/rfc/service-request.md

```diff
@@ -0,0 +1,228 @@
+# Service Request
+   
+## Overview and Motivation
+The Service handles incoming messages via several action Transports. On each incoming message Transport Router Adapter 
+builds an object with `ServiceRequest` type and passes it to the Dispatcher. There is no common constructor function now. 
+This object becomes available in the Action Handler as an argument. Its structure is abstract and functionality is 
+basically transport-agnostic.
+
+
+Most transport protocols natively have headers in messages. While currently every Transport Router Adapter is able to 
+parse incoming messages including its optional headers, in order to fully support messaging protocols the Service should 
+provide a way to set, modify and remove response message headers. That's why we need to implement **common reply headers 
+API**.
+
+
+Considering Node V8 hidden classes concept, to minimize hidden class trees number and respectfully maximize the 
+performance, the Service Request instance must always aim to have same object shape and its properties initialization 
+order. In order to make it even more performant we have to choose functions over ES6 classes for the Service Request 
+implementation. Due to that we need to use **single constructor function** to instantiate Service Request objects 
+anywhere.
+
+
+Developer experience could be improved in two ways: early headers validation and **strict and common validation policy**. 
+Early validation will help a developer to avoid errors on transport level. Since the limitations may be different, we 
+must comply with the strictest among all transports validation rules, so that any Action Handler could start to support 
+each of the Transports effortlessly.
+
+
+In order to respect backwards compatibility, we should be able to choose whether the reply will contain **only 
```

Comment:
I suggest `simpleResponse` options is not good design. Instead of this option I recommend to use an object with omitting undefined/empty values for 'headers' or 'data' 
`{ data? , headers? }`

Proc - No need to setup additional options. More clean call.
Cons - Breaking change for previous versions.

If backward compatibility is required, it should be handled in a special way in code.
To get the data use the workaround below:
`const actualData = response && response.data || response`